File: tests/fixtures/java/src/main/java/com/example/evenodd/Odd.java
package com.example.evenodd;

import static com.example.evenodd.Even.isEven;

/** Other half of the case d (D4) cross-file cycle. See Even.java. */
public class Odd {
    public static boolean isOdd(int n) {
        if (n == 0) {
            return false;
        }
        return isEven(n - 1);
    }
}

File: tests/fixtures/c-cpp/odd.cpp
#include "odd.h"

#include "even.h"

namespace odd {

// Other half of the case d (D4) cross-file cycle. See even.cpp.
bool is_odd(int n) {
    if (n == 0) {
        return false;
    }
    return even::is_even(n - 1);
}

}  // namespace odd
